### PR TITLE
Fixes #23278 - Migrate two-pane to breadcrumb-bar

### PR DESCRIPTION
**Pull request by sharvit**

### CODE REVIEW

File: app/views/discovery_rules/edit.html.erb [hunk 1]
```diff
@@ -1,3 +1,3 @@
-<% title _("Edit Discovery Rule") %>
+<% title _('Edit %s' % @discovery_rule.name) %>
```
Comment (lzap) [inline app/views/discovery_rules/edit.html.erb]: Must be `_('Edit %s') % @discovery_rule.name`

File: app/views/discovery_rules/_form.html.erb [hunk 2]
```diff
@@ -16,7 +16,7 @@
           <div class="tab-pane active" id="primary">
             <%= text_f f, :name %>
             <%= autocomplete_f f, :search,
-                               :path             => "discovered_hosts",
+                               :path             => "/discovered_hosts",
```
Comment (lzap) [inline app/views/discovery_rules/_form.html.erb]: Is this related? Can you explain?

**Review by lzap — CHANGES_REQUESTED** [review]
Thanks, can you give a bit of context? Is this happening in core? Mind appending a screenshot? @ik5 is also making a change atm, is this compatible?

I am working hard on getting tests fixed.

File: app/views/discovery_rules/index.html.erb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <% title _("Discovery Rules") %>
 <% title_actions new_link(_("Create Rule")), help_button %>
 
-<table class="<%= table_css_classes('table-two-pane') %>">
+<table class="<%= table_css_classes('table-fixed') %>">
```
Comment (ik5) [inline app/views/discovery_rules/index.html.erb:4]: Can you please explain why/what is that change and provide some screenshots of before and after?

**Comment by sharvit** [comment]
@lzap  @ik5 I am writing a small tutorial about how to migrate from `two-pane` (deprecated) to the new `BreadcrumbBar` so stay tuned :smile: 

First, remove the `table-two-pane` class to remove the `two-pane` functionality, see [two-pane.js](https://github.com/theforeman/foreman/blob/develop/app/assets/javascripts/two-pane.js)

Before (two-pane):
 
discoveryrules-two-pane

After (BreadcrumbBar):
discoveryrules-breadrumbbar

File: app/views/discovery_rules/edit.html.erb (hunk 1 above)
Comment (sharvit) [reply app/views/discovery_rules/edit.html.erb]: Sorry @lzap i'm not sure i understand, can you explain?

File: app/views/discovery_rules/_form.html.erb (hunk 2 above)
Comment (sharvit) [reply app/views/discovery_rules/_form.html.erb]: Yes, this is what lead to 404 error.

This form used to get rendered inside the `two-pane` view, which keep the index page url a the browser) and now it get rendered inside it own page with it own url `discovery_rules/new` or `discovery_rules/:id/edit`.
It makes the url wrong:
http://localhost:3000/discovery_rules/discovered_hosts/auto_complete_search?search=

Comment (sharvit) [reply app/views/discovery_rules/_form.html.erb]: I migrated it to use `discovered_hosts_path` :+1:

File: app/views/discovery_rules/edit.html.erb (hunk 1 above)
Comment (ik5) [reply app/views/discovery_rules/edit.html.erb]: @sharvit, you'ved made a format string, but called the format content inside the underscore function.

What @lzap wrote, is that you should do it on the result of the underscore function, and not as part of it's original string.

**Comment by ik5** [comment]
@sharvit that's cool :)

File: app/views/discovery_rules/edit.html.erb (hunk 1 above)
Comment (sharvit) [reply app/views/discovery_rules/edit.html.erb]: updated

**Comment by lzap** [comment]
Thanks, is there a core patch associated with this work? I don't see this in core yet.

**Comment by sharvit** [comment]
@lzap can you re-review please?

This might be helpful:
https://gist.github.com/sharvit/f7c975b16a289ec381673a7cc5c880e0

**Comment by ohadlevy** [comment]
@lzap bump?

**Comment by lzap** [comment]
Sorry had huge backlog, thanks it works.
